File: scripts/books.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys
from bs4 import BeautifulSoup
from lxml import etree

ifile = sys.argv[1]
ofile = sys.argv[2]

tree = etree.parse(ifile)

with open(ofile, 'w') as outf:
    outf.write(BeautifulSoup(
        etree.tostring(tree.find('body')), 'lxml'
    ).get_text())

File: scripts/books.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $idir = shift @ARGV;
my $odir = shift @ARGV;

my @files = `find $idir -type f -name "*.xml"`;

foreach my $ifile (@files) {
  print STDERR "Parsing $ifile";

  $ifile =~ m/.+\/([^\/]+)\.xml$/;
  my $ofile = $odir . "/" . $1 . ".txt";

  my $rc = system "python books.py $ifile $ofile";
  die "Error running books.py on $ifile: $!" if ($rc >> 8) != 0;
}
